src/app/containers/ConsentBanner/CanonicalLogic/index.test.js

```diff
@@ -0,0 +1,236 @@
+/* eslint-disable global-require */
+const PRIVACY_COOKIE = 'ckns_privacy';
+const EXPLICIT_COOKIE = 'ckns_explicit';
+const POLICY_COOKIE = 'ckns_policy';
+
+let Cookie;
+let setCookieOvenMock;
+const setShowPrivacyBannerMock = jest.fn();
+const setShowCookieBannerMock = jest.fn();
+
+const setCookieGetMock = ({ privacy = '1', explict = '1', policy = '111' }) => {
+  Cookie.get.mockImplementation(cookie => {
+    if (cookie === PRIVACY_COOKIE) {
+      return privacy;
+    }
+
+    if (cookie === EXPLICIT_COOKIE) {
+      return explict;
+    }
+
+    return policy;
+  });
+};
+
+const getConsentBannerUtilities = ({ logger } = {}) => {
+  const consentBannerUtilities = require('./index').default;
+  return consentBannerUtilities({
+    setShowPrivacyBanner: setShowPrivacyBannerMock,
+    setShowCookieBanner: setShowCookieBannerMock,
+    logger,
+  });
+};
+
+describe('Consent Banner Utilities', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  beforeEach(() => {
+    jest.mock('js-cookie', () => jest.fn());
+    Cookie = require('js-cookie');
+    Cookie.get = jest.fn();
+    Cookie.set = jest.fn();
+
+    jest.mock('./setCookieOven', () => jest.fn());
+    setCookieOvenMock = require('./setCookieOven');
+  });
+
+  describe('runInitial', () => {
+    it('does not show the privacy banner when PRIVACY_COOKIE is 1', () => {
+      setCookieGetMock({ privacy: '1' });
+
+      const { runInitial } = getConsentBannerUtilities();
+
+      runInitial();
+
+      expect(Cookie.set).toHaveBeenCalledTimes(0);
+      expect(setShowPrivacyBannerMock).not.toHaveBeenCalled();
+    });
+
+    it('sets PRIVACY_COOKIE and shows privacy banner when cookie is 0', () => {
+      setCookieGetMock({ privacy: '0' });
+
+      const { runInitial } = getConsentBannerUtilities();
+
+      runInitial();
+
+      expect(Cookie.set).toHaveBeenCalledTimes(1);
+      expect(Cookie.set).toHaveBeenCalledWith(PRIVACY_COOKIE, '1', {
+        expires: 365,
+      });
+      expect(setShowPrivacyBannerMock).toHaveBeenCalledWith(true);
+    });
+
+    it('sets PRIVACY_COOKIE and shows privacy banner when cookie is null', () => {
+      setCookieGetMock({ privacy: null });
+
+      const { runInitial } = getConsentBannerUtilities();
+
+      runInitial();
+
+      expect(Cookie.set).toHaveBeenCalledTimes(1);
+      expect(Cookie.set).toHaveBeenCalledWith(PRIVACY_COOKIE, '1', {
+        expires: 365,
+      });
+      expect(setShowPrivacyBannerMock).toHaveBeenCalledWith(true);
+    });
+
+    it('does not show the cookie banner when EXPLICIT_COOKIE is 1', () => {
+      setCookieGetMock({ explict: '1' });
```

Comment:
```suggestion
      setCookieGetMock({ explicit: '1' });
```

Bit worrying that this passes given there's a typo...?

Reply:
Assuming the typo is throughout :)

Reply:
Done :)